vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -439,6 +451,32 @@ type condition struct {
 	message       string
 }
 
+func (feeder *clusterStateFeeder) validateTargetRef(vpa *vpa_types.VerticalPodAutoscaler) (bool, []condition) {
+	if vpa.Spec.TargetRef == nil {
+		return true, []condition{}
+	}
+	k := controllerfetcher.ControllerKeyWithAPIVersion{
+		ControllerKey: controllerfetcher.ControllerKey{
+			Namespace: vpa.Namespace,
+			Kind:      vpa.Spec.TargetRef.Kind,
+			Name:      vpa.Spec.TargetRef.Name,
+		},
+		ApiVersion: vpa.Spec.TargetRef.APIVersion,
+	}
+	top, err := feeder.controllerFetcher.FindTopLevel(&k)
+	if err != nil {
+		return false, []condition{
+			{conditionType: vpa_types.ConfigUnsupported, delete: false, message: fmt.Sprintf("Error checking if target is a top level controller: %s", err)},
+		}
+	}
+	if top != nil && *top != k {
+		return false, []condition{
+			{conditionType: vpa_types.ConfigUnsupported, delete: false, message: "The targetRef controller has a parent"},
```

Comment:
What do you think about making the message more actionable? For example add "targetRef should point at a top-level controller"

Reply:
Done.